pkg/compiler/types.go

```diff
@@ -166,19 +174,21 @@ var typeArray = &typeDesc{
 				}
 			}
 			return &prog.BufferType{
-				TypeCommon: base.TypeCommon,
-				Kind:       bufKind,
-				RangeBegin: begin,
-				RangeEnd:   end,
+				TypeCommon:   base.TypeCommon,
```

Comment:
I think we should restrict this functionality only to ArrayType for now. I will make things in prog package simpler. When/if it will be needed for BufferType and we have it working for ArrayType, we can extend this to BufferType too. But I would avoid this unnecessary complexity for now.

Reply:
Reverted BufferType changes